runbotics/runbotics-desktop/src/action/jira/jira.utils.ts

```diff
@@ -131,27 +176,178 @@ export const getJiraUser = async <T extends CloudJiraUser | ServerJiraUser>({
+            .join(',');
+        dateCondition = `AND statusCategoryChangedDate in (${mappedDates})`;
+    }
+    const assignee = input?.email ? `AND assignee="${input.email}"` : '';
+    const status = input?.status ? `AND status="${input.status}"` : '';
+    const jql = `sprint="${input.sprint}" ${assignee} ${status} ${dateCondition}`.trim();
+
+    let startAt = 0;
+    const issues: SearchIssue<T>[] = [];
+
+    const response = await getJiraSprintTasksPage<T>(input, jql, startAt);
+    if (!response || !response.issues.length) {
+        throw new Error(`Tasks for sprint ${input.sprint} not found`);
+    }
+
+    startAt = response.maxResults;
+    issues.push(...response.issues);
+
+    while (response.total > startAt) {
+        const response = await getJiraSprintTasksPage<T>(input, jql, startAt);
+        issues.push(...response.issues);
+        startAt = response.startAt + response.maxResults;
+    }
+
+    return {
+        total: response.total,
+        issues,
+    };
+};
+
+const getJiraSprintTasksPage = async <T extends CloudJiraUser>(
+    input: GetSprintTasksInput,
+    jql: string,
+    startAt: Page['startAt'],
+) => {
+    const { data } =  await externalAxios.get<IssueWorklogResponse<T>>(
+        `${process.env[input.originEnv]}/rest/api/2/search`,
+        {
+            params: {
+                jql,
+                maxResults: 100, // 100 is max value
+                startAt,
+                fields: 'customfield_10016,timespent,duedate,statuscategorychangedate',
+            },
+            headers: getBasicAuthHeader(input),
+            maxRedirects: 0,
+        },
+    );
+
+    return data;
+};
+
+export const getJiraBoard = async (input: GetProjectSprintsInput) => {
+    const { data } =  await externalAxios.get<BoardResponse>(
+        `${process.env[input.originEnv]}/rest/agile/1.0/board`,
+        {
+            params: {
+                projectKeyOrId: input.project,
+                maxResults: 1,
+                startAt: 0,
+            },
+            headers: getBasicAuthHeader(input),
+            maxRedirects: 0,
+        },
+    );
+
+    if (!data || !data.values.length) {
+        throw new Error(`Board for project key or ID ${input.project} not found`);
+    }
+
+    return data.values[0];
+};
+
+export const getJiraAllBoardSprints = async (
+    boardId: Board['id'],
+    input: GetProjectSprintsInput,
+) => {
+    let startAt = 0;
+    const sprints: Sprint[] = [];
+
+    const response = await getJiraSprintPage(boardId, input, startAt);
+    if (!response || !response.values.length) {
+        throw new Error(`Sprints for board ID ${boardId} not found`);
+    }
+
+    startAt = response.maxResults;
+    sprints.push(...response.values);
+
+    while (response.total > startAt) {
+        const response = await getJiraSprintPage(boardId, input, startAt);
+        sprints.push(...response.values);
+        startAt = response.startAt + response.maxResults;
+    }
```

Comment:
You can do whatever you want in life, even become kungfu-monk who travels the word with his own, originally brewed combucha and bring a sense and taste into different people lifes, thanks to perfectly balanced mint and papaja extracts,

but don't use while loop.

please